### PR TITLE
[NP-7064] Proof-of-concept build script for FOAM

### PR DESCRIPTION
**Pull request by KernelDeimos**
Proof-of-concept build script for FOAM

### CODE REVIEW

File: src/foam/foobar/Exec.js [hunk 1]
```diff
@@ -0,0 +1,94 @@
...
+        spawnArgs[1] = [];
+      }
+      if ( this.passParentArgs ) spawnArgs[1].push(...this.parentArgs);
+      spawnArgs[1].push(...this.args);
+      console.log(`┏┫${spawnArgs[0]} ${spawnArgs[1].join(' ')}┃`);
+      const child = this.spawn_(...spawnArgs);
+      return await new Promise((rslv, rjct) => {
+        child.on('close', code => {
+          this.console.log(`┗┫EXIT ${code}┃\n`)
+          if ( code === 0 ) rslv(code);
```
Comment (kgrgreer) [inline src/foam/foobar/Exec.js]: Could you not format if/else's this way.

File: src/foam/foobar/FoobarConfig.js [hunk 2]
```diff
@@ -0,0 +1,91 @@
...
+ * Copyright 2022 The FOAM Authors. All Rights Reserved.
+ * http://www.apache.org/licenses/LICENSE-2.0
+ */
+
+foam.CLASS({
+  package: 'foam.foobar',
+  name: 'FoobarConfig',
+
+  nodeRequires: [
+    'path as path_',
```
Comment (kgrgreer) [inline src/foam/foobar/FoobarConfig.js]: no trailing commas

File: src/foam/foobar/FoobarTemplateUtil.js [hunk 3]
```diff
@@ -0,0 +1,102 @@
...
+          newline: function() {
+            self.push('\\n');
+          },
+          text: function(v) {
+            self.pushSimple(v);
+          }
+        });
+        return g;
+      }
+    },
```
Comment (kgrgreer) [inline src/foam/foobar/FoobarTemplateUtil.js]: trailing

File: src/foam/foobar/FoobarTemplateUtil.js [hunk 4]
```diff
@@ -0,0 +1,102 @@
...
+ * Copyright 2022 The FOAM Authors. All Rights Reserved.
+ * http://www.apache.org/licenses/LICENSE-2.0
+ */
+
+foam.CLASS({
+  package: 'foam.foobar',
+  name: 'FoobarTemplateUtil',
+  extends: 'foam.templates.TemplateUtil',
+
+  properties: [
```
Comment (kgrgreer) [inline src/foam/foobar/FoobarTemplateUtil.js:12]: Document the grammar to make it easier to understand than reading code.

File: src/foam/foobar/FoobarTemplateUtil.js [hunk 5]
```diff
@@ -0,0 +1,102 @@
+/**
+ * @license
+ * Copyright 2022 The FOAM Authors. All Rights Reserved.
+ * http://www.apache.org/licenses/LICENSE-2.0
+ */
+
+foam.CLASS({
```
Comment (kgrgreer) [inline src/foam/foobar/FoobarTemplateUtil.js:7]: Could FOAM's current templates work for this?

File: tools/deployJournals.js [hunk 6]
```diff
@@ -0,0 +1,77 @@
...
+
+if ( ! fs_.existsSync(outPath) ) {
+    fs_.mkdirSync(outPath, { recursive: true });
+}
+
+async function findJournals({ jrls, srcPath }) {
+    console.log('finding in ' + srcPath)
+    const walker = foam.util.filesystem.FileWalker.create();
+    walker.files.sub((_1, _2, info) => {
+        for ( const fileInfo of info.files ) {
```
Comment (kgrgreer) [inline tools/deployJournals.js]: indentation should be 2 spaces

File: src/foam/foobar/FoobarTemplateUtil.js (hunk 5 above)
Comment (KernelDeimos) [reply src/foam/foobar/FoobarTemplateUtil.js:7]: Tried that and almost pulled all my hair out. Can't remember why it didn't work. This grammar is mostly copied from the template parser.

Comment (KernelDeimos) [reply src/foam/foobar/FoobarTemplateUtil.js:7]: > Tried that and almost pulled all my hair out. Can't remember why it didn't work. This grammar is mostly copied from the template parser.

or at least, I think that's where it's from... wherever I pulled that from doesn't even exist anymore

**Pull request merged**